new: add `MarkdownLayoutProps` example to layout (#1503)
withastro/docs: src/pages/en/guides/markdown-content.md

</html>
```

> You can also add type safety using [the `Props` type](/en/guides/typescript/#component-props) with the `MarkdownLayoutProps` helper:

Reviewer: This is a great addition Ben, thank you!

Nit, but I wonder if “safety” is the best word here as we don’t actually do anything to check the layout gets passed the props it expects.

```suggestion
You can set a layout’s [`Props` type](/en/guides/typescript/#component-props) with the `MarkdownLayoutProps` helper:
```

```astro title="src/layouts/BaseLayout.astro" ins={2,4-9}
---
import type { MarkdownLayoutProps } from 'astro';

type Props = MarkdownLayoutProps<{
  // Define frontmatter props here
  title: string;
  author: string;
  date: string;
}>;

> // Now, `frontmatter`, `url`, and other markdown layout properties

Reviewer 2: ```suggestion
// Now, `frontmatter`, `url`, and other Markdown layout properties
```
Where do you think you are, MDX docs?? 😄 

Looks great, @bholmesdev! LGTM!
